File: leetcode/100-999/160. Intersection of Two Linked Lists/solution_2.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */

var getIntersectionNode = function(headA, headB) 
{
    const nodes = []
    let node = null
    
    node = headA
    while (node)
    {
        nodes.push(node)
        node = node.next
    }

    node = headB
    while (node)
    {
        if (nodes.includes(node))
            return node

        node = node.next
    }

    return null
};

File: leetcode/100-999/160. Intersection of Two Linked Lists/solution_2.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */

public class Solution
{
    public ListNode GetIntersectionNode(ListNode headA, ListNode headB)
    {
        HashSet<ListNode> nodes = new HashSet<ListNode>();
        ListNode node = null;

        node = headA;
        while (node != null)
        {
            nodes.Add(node);
            node = node.next;
        }

        node = headB;
        while (node != null)
        {
            if (nodes.Contains(node))
                return node;

            node = node.next;
        }

        return null;
    }
}